### PR TITLE
Add support for PSC nat and RLB proxy subnets

### PR DESCRIPTION
**Pull request by averbuks**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ All notable changes to this project will be documented in this file.
 - CloudSQ: fixed Terraform change detection when backup is disabled
 - Allow multiple CIDR blocks in the ip_range for Apigee Instance
 - Add prefix to project factory SA bindings 
+- Add support for Private Service Connect and Reginal Managed Proxy subnets for `net-vpc` module
```
Comment (aurelienlegrand) [inline CHANGELOG.md:11]: Typo: Reginal -> Regional Managed Proxy. Maybe explain that regional managed proxy subnets are for the new L7 Regional External Load Balancer?

File: modules/net-vpc/subnets.tf [hunk 2]
```diff
@@ -89,6 +89,14 @@ locals {
     for subnet in var.subnets_l7ilb :
     "${subnet.region}/${subnet.name}" => subnet
   }
+  subnets_l7rlb = {
```
Comment (aurelienlegrand) [inline modules/net-vpc/subnets.tf:92]: I think we should add an example of the net-vpc module using these 2 subnets ranges in the README.md file.

File: modules/net-vpc/subnets.tf [hunk 3]
```diff
@@ -142,6 +150,41 @@ resource "google_compute_subnetwork" "l7ilb" {
...
+  )
+  description = lookup(
+    local.subnet_descriptions,
+    "${each.value.region}/${each.value.name}",
+    "Terraform-managed."
+  )
+}
+
+resource "google_compute_subnetwork" "psc" {
+  provider      = google-beta
```
Comment (aurelienlegrand) [inline modules/net-vpc/subnets.tf:173]: Is google-beta mandatory here?

File: modules/net-vpc/subnets.tf [hunk 4]
```diff
@@ -142,6 +150,41 @@ resource "google_compute_subnetwork" "l7ilb" {
...
+  project       = var.project_id
+  network       = local.network.name
+  region        = each.value.region
+  name          = each.value.name
+  ip_cidr_range = each.value.ip_cidr_range
+  purpose       = "REGIONAL_MANAGED_PROXY"
+  role = (
+    each.value.active || each.value.active == null ? "ACTIVE" : "BACKUP"
+  )
+  description = lookup(
```
Comment (aurelienlegrand) [inline modules/net-vpc/subnets.tf:165]: Maybe mention in the description that this is a subnet for L7 regional XLB ?

File: modules/net-vpc/subnets.tf [hunk 5]
```diff
@@ -142,6 +150,41 @@ resource "google_compute_subnetwork" "l7ilb" {
...
+resource "google_compute_subnetwork" "psc" {
+  provider      = google-beta
+  for_each      = local.subnets_psc
+  project       = var.project_id
+  network       = local.network.name
+  region        = each.value.region
+  name          = each.value.name
+  ip_cidr_range = each.value.ip_cidr_range
+  purpose       = "PRIVATE_SERVICE_CONNECT"
+  description = lookup(
```
Comment (aurelienlegrand) [inline modules/net-vpc/subnets.tf:181]: Maybe mention in the description that this is a subnet for PSC NAT?

**Review by aurelienlegrand — COMMENTED** [review]
Small changes requested (mostly in the README.md file).

File: CHANGELOG.md (hunk 1 above)
Comment (averbuks) [reply CHANGELOG.md]: done

File: modules/net-vpc/subnets.tf (hunk 2 above)
Comment (averbuks) [reply modules/net-vpc/subnets.tf]: done

File: modules/net-vpc/subnets.tf (hunk 3 above)
Comment (averbuks) [reply modules/net-vpc/subnets.tf]: turns out it's not, removed

File: modules/net-vpc/subnets.tf (hunk 4 above)
Comment (averbuks) [reply modules/net-vpc/subnets.tf:141]: done

File: modules/net-vpc/subnets.tf (hunk 5 above)
Comment (averbuks) [reply modules/net-vpc/subnets.tf:156]: done

**Review by aurelienlegrand — APPROVED** [review]
Thanks for making the changes and for your contribution!

File: CHANGELOG.md (hunk 1 above)
Comment (aurelienlegrand) [reply CHANGELOG.md]: One last typo: "Private Service Connect and **Reginal** Managed" -> "Private Service Connect and **Regional** Managed"

Comment (averbuks) [reply CHANGELOG.md]: It is fixed

**Pull request merged**